torch/cuda/__init__.py

```diff
@@ -1137,6 +1137,46 @@ def _dtype(self):
 torch._storage_classes.add(ComplexDoubleStorage)
 torch._storage_classes.add(ComplexFloatStorage)
 
+
+class _WrappedTritonKernel(object):
+    """ Just a simple wrapper to store some metadata for testing purposes.
+    """
+
+    def __init__(self, kernel):
+        self.kernel = kernel
+        self.kernel_invoked = False
+
+    def __call__(self, *args, **kwargs):
+        res = self.kernel(*args, **kwargs)
+        self.kernel_invoked = True
+        return res
+
+
+def _register_triton_kernels():
+    import sys
+    import importlib
+
+    if sys.executable == "torch_deploy":
+        return
+
+    @_WrappedTritonKernel
+    def kernel_impl(*args, **kwargs):
+        from torch.sparse._triton_ops import bsr_dense_mm
+        return bsr_dense_mm(*args, skip_checks=True, **kwargs)
+
+    has_triton = importlib.util.find_spec("triton") is not None
+    if has_triton:
```

Comment:
Hmm, please help me understand the logic here: we would only register the kernel if somebody imported trition previously, which would not be the case for any of the eager workflows, isn't it?

Reply:
No, it just checks whether triton is importable.
```python
In [1]: import importlib, sys

In [2]: importlib.util.find_spec("triton") is None
Out[2]: False

In [3]: "triton" in sys.modules
Out[3]: False

```

Reply:
For context, this is basically what `import triton` does on the first time:
```python
spec = importlib.util.find_spec("triton")
triton = importlib.util.module_from_spec(spec)
spec.loader.exec_module(triton)
```

`find_spec` searches for the module, then `exec_module` is where the module's initialization code is actually run . So this checks that a module named "triton" exists, but doesn't run any code or guarantee that it will import successfully.